CMakeLists.txt

```
@@ -252,32 +256,51 @@ if(APPLE OR CMAKE_SYSTEM_NAME MATCHES "DragonFly|FreeBSD|Linux|NetBSD|OpenBSD")
   list(APPEND uv_test_libraries util)
 endif()
 
-add_library(uv SHARED ${uv_sources})
-target_compile_definitions(uv
+if(CMAKE_SYSTEM_NAME MATCHES "CYGWIN|MSYS")
```

CMake 3.4 doesn't support MSYS as a system name (only as a generator) but newer versions do. It would be easier to instead do

```suggestion
if(CYGWIN OR MSYS)
```

I should note that at some point I plan to turn all of these if blocks into generator expressions, so feel free to ignore this one.